hyperspace/cosmos/src/chain.rs

```diff
@@ -0,0 +1,111 @@
+use super::{error::Error, CosmosClient};
+use futures::Stream;
+use ibc_proto::{
+	cosmos::tx::v1beta1::{
+		mode_info::{Single, Sum},
+		AuthInfo, ModeInfo, SignDoc, SignerInfo, TxBody, TxRaw,
+	},
+	google::protobuf::Any,
+};
+use k256::ecdsa::{signature::Signer as _, Signature, SigningKey};
+use primitives::{Chain, IbcProvider};
+use prost::Message;
+use std::pin::Pin;
+use tendermint_rpc::Client;
+
+#[async_trait::async_trait]
+impl<H> Chain for CosmosClient<H>
+where
+	H: Clone + Send + Sync + 'static,
+{
+	fn name(&self) -> &str {
+		&*self.name
+	}
+
+	fn block_max_weight(&self) -> u64 {
+		todo!()
+	}
+
+	async fn estimate_weight(&self, messages: Vec<Any>) -> Result<u64, Self::Error> {
+		todo!()
+	}
+
+	async fn finality_notifications(
+		&self,
+	) -> Pin<Box<dyn Stream<Item = <Self as IbcProvider>::FinalityEvent> + Send + Sync>> {
+		todo!()
+			FinalityProtocol::Tendermint => {
```

Comment:
I meant to remove the entire match statement, that was here previously

Reply:
Oh, my bad. Fixed [34894d1](https://github.com/ComposableFi/centauri/pull/114/commits/34894d1187d07d6eba6c891be26b3e20f96983d0)